Encounter build error caused by redefinition of declarations on Windows

**Describe the bug**
A build error caused by redefinitions of `c macro` and `function declarations`.

**To Reproduce**
Steps to reproduce the behavior:
1. cd samples\hello_world
2. set BOARD=npcx9m6f_evb
3. west build -p always
4. See error

**Expected behavior**
build successfully.

**Logs and console output**
Please refer attached [build_error.zip](https://github.com/zephyrproject-rtos/zephyr/files/11784865/build_error.zip). We found lots of  redefined warnings/errors in `syscall_list.h` and `kernel.h` generated by `gen_syscalls.py`
```
In file included from C:/Users/TomChang/Desktop/zephyrproject/zephyr/include/zephyr/syscall.h:11,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/include/zephyr/kernel_includes.h:31,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/arch/arm/core/offsets/offsets.c:9:
C:/Users/TomChang/Desktop/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/syscall_list.h:215: warning: "K_SYSCALL_CLOCK_GETTIME" redefined
  215 | #define K_SYSCALL_CLOCK_GETTIME 206
      | 
C:/Users/TomChang/Desktop/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/syscall_list.h:6: note: this is the location of the previous definition
    6 | #define K_SYSCALL_CLOCK_GETTIME 0
```

**Environment (please complete the following information):**
 - OS:  Windows
 - Toolchain - Zephyr SDK, 15.1
 - Commit SHA: 32fae07f299aa326

--- a/scripts/build/parse_syscalls.py
+++ b/scripts/build/parse_syscalls.py
@@ -29,6 +29,7 @@ import re
 import argparse
 import os
 import json
+from pathlib import PurePath
 
 regex_flags = re.MULTILINE | re.VERBOSE
 
@@ -100,6 +101,8 @@ def analyze_headers(include_dir, scan_dir, file_list):
                         path.endswith(os.path.join(os.sep, 'toolchain',
                                                    'common.h'))):
                     continue
+
+                path = PurePath(os.path.normpath(path)).as_posix()
 
                 if path not in syscall_files:
                     if include_dir and base_path in include_dir:

